src/Plutus/Model/Validator.hs

```diff
@@ -104,6 +104,20 @@ instance HasValidator (TypedValidator datum redeemer) where
 instance HasAddress (TypedValidator datum redeemer) where
   toAddress = toAddress . toValidatorHash
 
+-- untyped validator
+
+instance HasDatum (Versioned Validator) where
+  type DatumType (Versioned Validator) = Datum
+
+instance HasRedeemer (Versioned Validator) where
```

Comment:
I like that you've also added the `HasRedeemer` and `HasDatum` for the untyped versions 👌